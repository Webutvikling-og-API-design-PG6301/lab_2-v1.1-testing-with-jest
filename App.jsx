import React, {useState} from "react";
import {Link, Route, Routes, useNavigate} from "react-router-dom";
import {randomQuestion, isCorrectAnswer} from "./questions";

export function FrontPage({correctAnswer, questionsAnswered}) {
    return <div data-testid={"status"}>
            <h1>Welcome to the most simplified quiz ever!!</h1>
            <h2>you have answered {correctAnswer} / {questionsAnswered} correct</h2>
       <Link to={"/showQuestion"}>
           <button>ny quiz</button>
       </Link>
        </div>

}

function ShowQuestion({setCorrectAnswer, setQuestionAnswered}) {
    const [question, setQuestion] = useState(randomQuestion())
    const navigate = useNavigate()
    function handleAnswer( answer) {
        setQuestionAnswered(prev => prev +1)
        if(isCorrectAnswer(question, answer)) {
            setCorrectAnswer(prev => prev +1)
            navigate("/answer/correct")
        } else if(!isCorrectAnswer(question, answer)) {
            navigate("/answer/wrong")
        }

    }
    return<div>
            <h1>Welcome to show question</h1>
            <h2>{question.question}</h2>
            {Object.keys(question.answers).filter(value => question.answers[value]).map(display => {

                return <div key={display}> <button onClick={() => handleAnswer(display)}>{question.answers[display]}</button></div>
            })}
        </div>

}


function ShowAnswer() {
    return <div>
        <Routes>
            <Route path={"correct"} element={<h1>correct</h1>} />
            <Route path={"wrong"} element={<h1>wrong</h1>} />
        </Routes>
        <div>
            <Link to={"/"}>Show score</Link>
        </div>
        <div>
            <Link to={"/showQuestion"}>new question</Link>
        </div>
    </div>;
}

export function App() {
    const [correctAnswer, setCorrectAnswer] = useState(0)
    const [questionsAnswered, setQuestionAnswered] = useState(0)
    console.log(correctAnswer)
    return(
        <Routes>
            <Route path={"/"} element={<FrontPage correctAnswer={correctAnswer} questionsAnswered={questionsAnswered} />}/>
            <Route path={"/showQuestion"} element={<ShowQuestion setCorrectAnswer={setCorrectAnswer} setQuestionAnswered={setQuestionAnswered} />} />
            <Route path={"/answer/*"} element={<ShowAnswer />} />
        </Routes>
    )
}
