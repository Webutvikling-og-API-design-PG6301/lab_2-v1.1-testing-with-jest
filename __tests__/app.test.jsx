import React from 'react'
import {render} from "react-dom";
import {FrontPage} from "../App";

describe("Frontpage", () => {

    it("should be shown", () => {
        const element = document.createElement("div");
        render(<FrontPage />, element)
        expect(element.innerHTML).toMatchSnapshot()
    })
})