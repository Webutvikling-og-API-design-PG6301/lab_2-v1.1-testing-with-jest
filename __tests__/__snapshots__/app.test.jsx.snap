// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontpage should make all component to snapshot 1`] = `"<div data-testid=\\"status\\"><h1>Welcome to the most simplified quiz ever!!</h1><h2>you have answered  /  correct</h2><a href=\\"/showQuestion\\"><button>ny quiz</button></a></div>"`;
